tftb/processing/cohen.py

```diff
@@ -263,29 +263,37 @@ def smoothed_pseudo_wigner_ville(signal, timestamps=None, freq_bins=None,
         taumax = min([ti + lg - 1, signal.shape[0] - ti + lg,
                       np.round(freq_bins / 2.0) - 1, lh])
         points = np.arange(-min([lg, signal.shape[0] - ti]),
-                           min([lg, ti - 1]) + 1)
-        g2 = twindow[lg + points]
+                           min([lg, ti - 1]) + 1).astype(int)
+        lg_points = [int(i) for i in (lg + points)]
+        g2 = twindow[lg_points]
         g2 = g2 / np.sum(g2)
-        tfr[0, icol] = np.sum(g2 * signal[ti - points - 1] * np.conj(signal[ti - points - 1]))
+        signal_idx = [int(i) for i in (ti - points - 1)]
+        tfr[0, icol] = np.sum(g2 * signal[signal_idx] * np.conj(signal[signal_idx]))
         for tau in range(int(taumax)):
             points = np.arange(-min([lg, signal.shape[0] - ti - tau]),
                                min([lg, ti - 1 - tau]) + 1)
-            g2 = twindow[lg + points]
+            lg_points = [int(i) for i in (lg + points)]
+            g2 = twindow[lg_points]
             g2 = g2 / np.sum(g2)
-            R = np.sum(g2 * signal[ti + tau - points - 1] * np.conj(signal[ti - tau - points - 1]))
-            tfr[1 + tau, icol] = fwindow[lh + tau + 1] * R
-            R = np.sum(g2 * signal[ti - tau - points - 1] * np.conj(signal[ti + tau - points - 1]))
-            tfr[freq_bins - tau - 1, icol] = fwindow[lh - tau + 1] * R
+            idx1 = [int(i) for i in (ti + tau - points - 1)]
+            idx2 = [int(i) for i in (ti - tau - points - 1)]
+            R = np.sum(g2 * signal[idx1] * np.conj(signal[idx2]))
+            tfr[1 + tau, icol] = fwindow[int(lh + tau + 1)] * R
+            R = np.sum(g2 * signal[idx2] * np.conj(signal[idx1]))
+            tfr[freq_bins - tau - 1, icol] = fwindow[int(lh - tau + 1)] * R
         tau = np.round(freq_bins / 2.0)
         if (ti <= signal.shape[0] - tau) and (ti >= tau + 1) and (tau <= lh):
             points = np.arange(-min([lg, signal.shape[0] - ti - tau]),
                                min([lg, ti - 1 - tau]) + 1)
-            g2 = twindow[lg + 1 + points]
+            lg_points = [int(i) for i in (lg + 1  + points)]
+            g2 = twindow[lg_points]
             g2 = g2 / np.sum(g2)
-            _x = np.sum(g2 * signal[ti + tau - points] * np.conj(signal[ti - tau - points]))
-            _x *= fwindow[lh + tau + 1]
-            _y = np.sum(g2 * signal[ti - tau - points] * np.conj(signal[ti + tau - points]))
-            _y *= fwindow[lh - tau + 1]
+            idx1 = [int(i) for i in (ti + tau - points)]
+            idx2 = [int(i) for i in (ti - tau - points)]
+            _x = np.sum(g2 * signal[idx1] * np.conj(signal[idx2]))
+            _x *= fwindow[int(lh + tau + 1)]
+            _y = np.sum(g2 * signal[idx2] * np.conj(signal[idx1]))
+            _y *= fwindow[int(lh - tau + 1)]
             tfr[tau, icol] = (_x + _y) * 0.5
```

Comment:
Instead of using list comprehensions, as @dan-1d pointed out, we could simply use `array.astype(int)` wherever necessary. Can you please use that in the appropriate places?

Reply:
Of course! I will work on it in the next few days.